### PR TITLE
adds External avatars

### PR DESCRIPTION
**Pull request by raftx24**

### CODE REVIEW

File: database/migrations/2017_01_01_113000_create_avatars_table.php [hunk 1]
```diff
@@ -15,6 +15,9 @@ public function up()
             $table->foreign('user_id')->references('id')->on('users')
                 ->onUpdate('cascade')->onDelete('cascade');
 
+            $table->tinyInteger('type')->unsigned();
```
Comment (aocneanu) [inline database/migrations/2017_01_01_113000_create_avatars_table.php:18]: I guess the url would be enough? If the url is not null it's gravatar, if it's null it has a local file

File: src/Services/Generators/Gravatar.php [hunk 2]
```diff
@@ -2,22 +2,24 @@
 
 namespace LaravelEnso\Avatars\Services\Generators;
 
-use Illuminate\Http\File;
 use Illuminate\Support\Facades\Http;
-use Illuminate\Support\Facades\Storage;
+use LaravelEnso\Avatars\Enums\Types;
 use LaravelEnso\Avatars\Models\Avatar;
 
 class Gravatar extends Generator
```
Comment (aocneanu) [inline src/Services/Generators/Gravatar.php:9]: let's drop the generator class and use constructors in both gravatar and laravolt

File: src/Services/Generators/Laravolt.php [hunk 3]
```diff
@@ -5,22 +5,45 @@
...
 {
     private const FontSize = 128;
+    private const Filename = 'avatar';
+    private const Extension = 'jpg';
+    private string $path;
 
-    public function generate(): File
+    public function handle(): ?Avatar
+    {
+        $this->generate()->attach();
```
Comment (aocneanu) [inline src/Services/Generators/Laravolt.php:22]: let's do generate->persist->attach

right now attach also persists the avatar model, extract it

File: database/migrations/2017_01_01_113000_create_avatars_table.php (hunk 1 above)
Comment (aocneanu) [reply database/migrations/2017_01_01_113000_create_avatars_table.php:18]: @raftx24

File: src/Enums/Types.php [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace LaravelEnso\Avatars\Enums;
+
+class Types
+{
+    public const File = 1;
+    public const External = 2;
```
Comment (aocneanu) [inline src/Enums/Types.php:8]: File
Url

Comment (raftx24) [reply src/Enums/Types.php:8]: `Types` was removed

**Pull request merged**